assets/javascripts/discourse/initializers/require-tag-on-topic-creation.js

```diff
@@ -0,0 +1,32 @@
+import bootbox from "bootbox";
+import I18n from "I18n";
+import { withPluginApi } from "discourse/lib/plugin-api";
+
+export default {
+  name: "require-tag-on-topic-creation",
+
+  initialize() {
+    withPluginApi("1.3.0", (api) => {
+      api.composerBeforeSave(() => {
+        return new Promise((resolve, reject) => {
+          const composerModel = api.container.lookup(
+            "controller:composer"
+          ).model;
+          const globalFilters = api.container
+            .lookup("site-settings:main")
+            .global_filters.split("|");
+
+          if (
+            composerModel.tags.filter((tag) => globalFilters.includes(tag))
+              .length > 0
+          ) {
+            resolve();
+          } else {
+            bootbox.alert(I18n.t("global_filter.require_tag_on_topic_creation.error"));
```

Comment:
Oh, let's use the new dialog service, bootbox is soon-to-be deprecated. 

Something like 

```
import { getOwner } from "discourse-common/lib/get-owner";

const dialog = getOwner(this).lookup("service:dialog");
this.dialog.alert(msg);

```